File: examples-sdk/zig/009-basic-credit-regulatory-slik/main.zig
// 009-basic-credit-regulatory-slik — Zig SDK equivalent
// Compile: vil compile --from zig --input 009-basic-credit-regulatory-slik/main.zig --release

const vil = @import("vil");

pub fn main() void {
    var p = vil.Pipeline.init("RegulatoryStreamPipeline", 3083);
    p.sink("regulatory_sink", 3083, "/regulatory-stream");
    p.source("regulatory_source", "http://localhost:18081/api/v1/credits/ndjson?count=1000", "json");
    p.route("sink.trigger_out", "source.trigger_in", "LoanWrite");
    p.route("source.response_data_out", "sink.response_data_in", "LoanWrite");
    p.route("source.response_ctrl_out", "sink.response_ctrl_in", "Copy");
    p.compile();
}

File: examples-sdk/go/009-basic-credit-regulatory-slik/main.go
// 009-basic-credit-regulatory-slik — Go SDK equivalent
// Compile: vil compile --from go --input 009-basic-credit-regulatory-slik/main.go --release
package main

import vil "github.com/OceanOS-id/vil-go"

func main() {
	p := vil.NewPipeline("RegulatoryStreamPipeline", 3083)

	p.Sink(vil.SinkOpts{Name: "regulatory_sink", Port: 3083, Path: "/regulatory-stream"})
	p.Source(vil.SourceOpts{Name: "regulatory_source", URL: "http://localhost:18081/api/v1/credits/ndjson?count=1000", Format: "json"})

	p.Route("sink.trigger_out", "source.trigger_in", "LoanWrite")
	p.Route("source.response_data_out", "sink.response_data_in", "LoanWrite")
	p.Route("source.response_ctrl_out", "sink.response_ctrl_in", "Copy")

	p.Compile()
}
